### PR TITLE
chore: fix build on gcc 10

### PR DESCRIPTION
**Pull request by glennsl**
OCaml 4.8 doesn't build with gcc 10, so this bumps it to 4.10 in dev, and pins opam to 2.5.0 to get rid of the secondary compiler, which is also 4.8.

### CODE REVIEW

File: brisk-reconciler.opam [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
```
Comment (glennsl) [inline brisk-reconciler.opam:26]: Not sure what this is about. All I've done is tweak the dependencies, the dune-project `lang`, and run `esy`.

**Comment by glennsl** [comment]
:( caching on the windows CI box might be broken

File: brisk-reconciler.opam (hunk 1 above)
Comment (wokalski) [reply brisk-reconciler.opam:26]: New dune version 😄

**Comment by Et7f3** [comment]
I think we can erase this devDeps on dune2.5.0 because ocaml@4.8 has received the proper gcc with upstream

**Comment by glennsl** [comment]
Indeed it has! Submitted #82 to remove it. Thanks for the heads up @Et7f3

**Pull request merged**